### PR TITLE
Keep screen on when viewing a video

### LINKED ISSUE #6642: Screen turns off during video playback
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
During video playback the screen turns off. There should be a wakelock preventing that the screen turns off when a video is playing.

### Steps to reproduce
- set turn off timer in Android settings to a short time (e.g. 15s)
- go to a conversation with a video attachment, that plays longer than the turn off time of the display
- tap this video
- wait as long as your display turn off time is (don't interact with the screen)

**Actual result:** Screen turns off and interrupts video playback.
**Expected result:** Screen stays on during video playback.

### Device info 1

**Device:** Fairphone 2
**Android version:** 6.0.1
**Signal version:** 4.6.0

### Device info 2
**Device:** Sony Xperia U
**Android version:** 4.4.4 (CyanogenMod)
**Signal version:** 4.6.0

### PR DESCRIPTION
**Pull request by raffthesweetling13**
Fixes #6642

// FREEBIE

### First time contributor checklist

- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Virtual device: Nexus 6, Android 6.0
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description

This change ensures that the screen remains on when a video is playing on devices running SDK >= 16. The screen will also turn off after after the normal sleep time if the video has been paused. 

**Testing**

Test setup: Sent myself a 40 second long video, and set the screen timeout to 15 seconds.

1. Start video from message list, and entire video plays without the screen shutting off. 15 seconds after the video completes, the screen shuts off.
2. Pause video and screen shuts off after 15 seconds. 
3. Pause video and seek, then screen shuts off after 15 seconds. 

I'd also like some feedback on a couple things:
1. The cast [on this line](https://github.com/raffthesweetling13/Signal-Android/commit/7294b5ec528e767bba22c4a3c332cb0bedb9cd6e#diff-3b227dc2e0ee4d3f8a6a8a4702a1c3d3R118)
2. Having a class that implements an interface, but leaves most of the methods blank

### CODE REVIEW

**Comment by FeuRenard** [comment]
First, thanks for tackling this issue! 👍 
Did you test the following?

- Start video from conversation, immediately pause, resume play.

> I'd also like some feedback on a couple things:
> 1. The cast on this line

Moxie said [here](https://github.com/WhisperSystems/Signal-Android/pull/6253#discussion_r104533174): "hmm i don't really like this cast". Personally, I would add a public method `addEventListener(ExoPlayer.EventListener eventListener)` to `VideoPlayer` and call this method from `MediaPreviewActivity`, maybe [here](https://github.com/WhisperSystems/Signal-Android/blob/0ae60da68b86efd5916c27984e474f5e1b39c797/src/org/thoughtcrime/securesms/MediaPreviewActivity.java#L179). So you would get rid of the cast.

> 2. Having a class that implements an interface, but leaves most of the methods blank

You could use an [anonymous class](http://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html), but I think even then you have to implement every method. I would try a nested class in `MediaPreviewActivity` in combination what I wrote above.

But please note: I'm not this project's maintainer. Changes I like don't have to match Moxie's taste.

File: src/org/thoughtcrime/securesms/video/exo/ExoPlayerListener.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        break;
+      default:
+        break;
+    }
+  }
+
+  @Override
+  public void onTimelineChanged(Timeline timeline, Object manifest) {
+
+  }
```
Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/video/exo/ExoPlayerListener.java]: I would say, you can omit two newlines here and for the following methods. Just: `public void onTimelineChanged(Timeline timeline, Object manifest) { }` (in one line)

**Comment by raffthesweetling13** [comment]
Thanks for the feedback!

I like you suggestion of making the listener a nested class, but I felt like it should be nested in `VideoPlayer` instead of `MediaPreviewActivity` since the listener only applies to videos. Then to avoid the cast I passed the `Window` in from `MediaPreviewActivity` to `VideoPlayer` so that `VideoPlayer` could use it when creating the listener. 

I also removed the extra new lines from the no-op methods of the listener. 

I retested all the test cases I mentioned above, and the one you suggested.

**Comment by moxie0** [comment]
in 4.10.0

**Pull request closed without merging**